docs/how-to/incremental-migrations.md

```diff
@@ -0,0 +1,365 @@
+    );
+    update_character(
+        ctx,
+        Character {
+            level: character.level + 1,
+            ..character
+        },
+    );
+}
+```
+
+We'll play around a bit with `spacetime call` to set up a character:
+
+```sh
+$ spacetime logs incr-migration-demo -f &
+
+$ spacetime call incr-migration-demo create_character '{ "Fighter": {} }' "Phoebe"
+
+2025-01-07T15:32:57.447286Z  INFO: src/lib.rs:21: Creating new level 1 Fighter named Phoebe
+
+$ spacetime call -s local incr-migration-demo rename_character "Gefjon"
+
+2025-01-07T15:33:48.966134Z  INFO: src/lib.rs:48: Renaming Phoebe to Gefjon
+
+$ spacetime call -s local incr-migration-demo level_up_character
+
+2025-01-07T15:34:01.437495Z  INFO: src/lib.rs:66: Leveling up Gefjon from 1 to 2
+
+$ spacetime sql incr-migration-demo 'SELECT * FROM character'
+
+ player_id | nickname | level | class
+-----------+----------+-------+----------------
+ <snip>    | "Gefjon" | 2     | (Fighter = ()) 
+```
+
+Now we want to add a new feature: each player should be able to align themselves with the forces of good or evil, so we can get some healthy competition going between our players. We'll start each character off with `Alliance::Neutral`, and then offer them a reducer `choose_alliance` to set it to either `Alliance::Good` or `Alliance::Evil`. Our first attempt will be to add a new column to the type `Character`:
+
+```rust
+#[spacetimedb::table(name = character, public)]
+struct Character {
+    #[primary_key]
+    player_id: Identity,
+    nickname: String,
+    level: u32,
+    class: Class,
+    alliance: Alliance,
+}
+
+#[derive(SpacetimeType, Debug, Copy, Clone)]
+enum Alliance {
+    Good,
+    Neutral,
+    Evil,
+}
+
+#[spacetimedb::reducer]
+fn choose_alliance(ctx: &ReducerContext, alliance: Alliance) {
+    let character = find_character_for_player(ctx);
+    log::info!(
+        "Setting {}'s alliance to {:?} for player {}",
+        character.nickname,
+        alliance,
+        ctx.sender,
+    );
+    update_character(
+        ctx,
+        Character {
+            alliance,
+            ..character
+        },
+    );
+}
+```
+
+But that will fail, since SpacetimeDB doesn't know how to update our existing `character` rows with the new column:
+
+```
+Error: Database update rejected: Errors occurred:
+Adding a column alliance to table character requires a manual migration
+```
+
+Instead, we'll add a new table, `character_v2`, which will coexist with our original `character` table:
+
+```rust
+#[spacetimedb::table(name = character_v2, public)]
+struct CharacterV2 {
+    #[primary_key]
+    player_id: Identity,
+    nickname: String,
+    level: u32,
+    class: Class,
+    alliance: Alliance,
+}
+```
+
+When a new player creates a character, we'll make rows in both tables for them. This way, any old clients that are still subscribing to the original `character` table will continue to work, though of course they won't know about the character's alliance.
```

Comment:
You know, we may also want a tutorial on writing ECS-style code. That paradigm fits into SQL well enough and would achieve what this guide achieves. (You still want both guides because in some cases you need this.) Unfortunately, we don't have good support for the foreign-key constraints you'd ideally want when writing ECS code.